### PR TITLE
be able to install admincontroller webhook in namespace other than 'd…

### PR DESCRIPTION
**Pull request by pruan-rht**
…efault'

### CODE REVIEW

**Comment by snir911** [comment]
Thanks for that,
Look good to me, I'd only split or change the first commit subject to be < 70
I also wonder whether deploy/webhook.yaml should also be generated/changed on the fly now, to match the namespace.

**Comment by wainersm** [comment]
> Look good to me, I'd only split or change the first commit subject to be < 70

Yes. It needs to follow this: https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format

> I also wonder whether deploy/webhook.yaml should also be generated/changed on the fly now, to match the namespace.

The commit https://github.com/kata-containers/tests/pull/3091/commits/2e8ad313734b845ff18e6dd536136a9bb33d86ab will end up generating webhook.yaml with a namespace property.

**Comment by wainersm** [comment]
> The commit [2e8ad31](https://github.com/kata-containers/tests/commit/2e8ad313734b845ff18e6dd536136a9bb33d86ab) will end up generating webhook.yaml with a namespace property.

Never mind the above comment, webhook.yaml is indeed not generated and @snir911 is correct.

File: kata-webhook/create-certs.sh [hunk 1]
```diff
@@ -7,22 +7,22 @@
...
+kubectl create -n ${WEBHOOK_NS} secret generic \
     ${WEBHOOK_SVC}-certs \
     --from-file=key.pem=./webhookCA.key \
     --from-file=cert.pem=./webhook.crt \
     --dry-run -o yaml > ./deploy/webhook-certs.yaml
 
 # Set the CABundle on the webhook registration
 CA_BUNDLE=$(cat ./webhook.crt | base64 -w0)
-sed "s/CA_BUNDLE/${CA_BUNDLE}/" ./deploy/webhook-registration.yaml.tpl > ./deploy/webhook-registration.yaml
+sed -e "s/CA_BUNDLE/${CA_BUNDLE}/"  -e "s/PROJECT_NAMESPACE/${WEBHOOK_NS}/" ./deploy/webhook-registration.yaml.tpl > ./deploy/webhook-registration.yaml
```
Comment (jodh-intel) [inline kata-webhook/create-certs.sh:24]: This would be clearer if you folded the line a few times with backslashes.

**Review by jodh-intel — COMMENTED** [review]
Thanks for raising @pruan-rht. However, we require that PRs are formatted in a particular way.

Please take a look at our [contributing guide](https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format) which explains what you have to do to make the PR "well formed", which will allow the CI's to pass (some are currently failing if you look at the bottom of the PR).

**Comment by fidencio** [comment]
@pruan-rht, This is a valid change but it's been stale for quite a long time.  I'd like to ask you to resubmit against the `main` branch, as that's where the current development is happening.

The main reason behind closing this one is helping to have a better idea on what we really need for the final 1.13.0 release.  If you think this is still something that **must** go on for 1.13.0, please, open the PR, update it accordingly and send me a message (just `@fidencio` here).

**Pull request closed without merging**